optax/contrib/_ademamix.py

```diff
@@ -0,0 +1,182 @@
+"""AdeMAMix.
+
+Implementation of
+"THE ADEMAMIX OPTIMIZER: BETTER, FASTER, OLDER"
+(https://arxiv.org/pdf/2409.03137) by Matteo Pagliardini, 
+Pierre Ablin and David Grangier.
+"""
+import optax.tree_utils as otu
+from typing import NamedTuple, Optional
+import chex
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from optax._src import base
+from optax._src import combine
+from optax._src import numerics
+from optax._src import transform
+
+
+class ScaleByAdemamixState(NamedTuple):
+  """State for the Ademamix algorithm."""
+
+  count: chex.Array
+  count_m2: chex.Array
```

Comment:
I'm not sure why you have multiple counts. Would just the first suffice?

Reply:
So the first `count` replicates the same entitiy in `adam`.  The second is for keeping track of the second momentum, so the user might want to do different things with each.

Does that make sense?

I will make an inline comment to this fact